### ISSUE TITLE
Possibly a Universal/Cross-Language Port

### ISSUE THREAD

**Issue by MoralCode** [OpenRowingCommunity/c2bluetooth#21]
A super long term "eventually" goal is to make this library cross platform across other languages using a technique similar to this:

https://www.osohq.com/post/cross-platform-rust-libraries

**Comment by MonsieurRz** [comment]
Idk at the time of the writing but there is dart ffi now.
Interestingly, during the dart, rust and c ffi demo, dart was still very competitive (with some minor code changes : 
[https://youtu.be/HF8gHAakb1Q?si=XB1rb529z3gfBmFE]()

**Comment by MoralCode** [comment]
yeah i think i would want to see this library get a lot more usage or demand  before investing too much into cross-platformness
